### ISSUE TITLE
Tidy should read files in a better way...

### ISSUE THREAD

**Issue by Wafflespeanut** [servo/servo#9060]
Currently, our tidy gets the contents of all files using `file.read()` function, and we have 8 functions to which the contents are passed, out of which 5 functions apply `splitlines` to slice the contents. This is bad, because we're traversing through the string multiple times *unnecessarily* for achieving that. As an alternative, we can make use of the `file.readlines` (awesomesauce!) to pass the list of lines to the functions directly. Three of those functions require the entire contents, for which we could join the list of lines. This won't be expensive since there aren't much webidl, lock & py files in the repo.

Code: `python/tidy.py`

**Comment by dsprenkels** [comment]
I would like to work on this, to see if the servo project fits me. :)

**Comment by KiChjang** [comment]
Alright!

**Comment by dsprenkels** [comment]
I have built a fix, but it does not make any difference in performance. The string handling does not take any time currently.
Running `$ python -m cProfile mach test-tidy | grep "{method 'splitlines' of 'str' objects}\|{method 'join' of 'str' objects}"`:

before patch:
```
   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
    11102    0.007    0.000    0.008    0.000 {method 'join' of 'str' objects}
     1909    0.031    0.000    0.031    0.000 {method 'splitlines' of 'str' objects}
```

after patch:
```
    11948    0.012    0.000    0.012    0.000 {method 'join' of 'str' objects}
      395    0.001    0.000    0.001    0.000 {method 'splitlines' of 'str' objects}
```

So the difference is about 36 milliseconds, when testing the whole code base. Would you like me to submit the [patch](https://github.com/servo/servo/compare/master...dsprenkels:issue-9060?expand=1#diff-bd72e3b03cbd2b983d260b9425f80d46)?

**Comment by Wafflespeanut** [comment]
I'm surprised that there's not much improvement. Please go ahead and open a PR. Let's try to reduce the time further :)

**Comment by Wafflespeanut** [comment]
fixed by #9066
